server/metrics/metrics.go

```diff
@@ -109,6 +111,18 @@ func NewMetrics(info InstanceInfo) Metrics {
 	)
 	m.registry.MustRegister(m.apiTime)
 
+	m.messageSyncTime = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace:   MetricsNamespace,
+			Subsystem:   MetricsSubsystemApp,
+			Name:        "message_sync_time",
```

Comment:
Should this be named `messages_delay_seconds` as per https://docs.google.com/document/d/1CFGMXUg10oZeoNm_rduLoVIwi0SRiHxj529NJaYtEjc/edit? Note the trailing unit as per Prometheus naming conventions: https://prometheus.io/docs/practices/naming/.

Open to proposing a rename, but want to make sure we're aligned on the meaning first.